File: src/src/main/java/com/mewebstudio/translatable/entity/Category.java
package com.mewebstudio.translatable.entity;

import com.mewebstudio.springboot.jpa.translatable.ITranslatable;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "categories")
@Getter
@Setter
@NoArgsConstructor
public class Category extends AbstractBaseEntity implements ITranslatable<String, CategoryTranslation> {
    @OneToMany(mappedBy = "owner", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("locale ASC")
    private List<CategoryTranslation> translations = new ArrayList<>();

    @Override
    public String toString() {
        return this.getClass().getSimpleName() + "{" +
            "id='" + getId() + '\'' +
            ", createdAt=" + getCreatedAt() +
            ", updatedAt=" + getUpdatedAt() +
            '}';
    }
}

File: src/main/kotlin/com/mewebstudio/translatable/entity/Category.kt
package com.mewebstudio.translatable.entity

import com.mewebstudio.springboot.jpa.translatable.kotlin.ITranslatable
import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.OneToMany
import jakarta.persistence.OrderBy
import jakarta.persistence.Table

@Entity
@Table(name = "categories")
class Category(
    @OneToMany(mappedBy = "owner", cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("locale ASC")
    override var translations: MutableList<CategoryTranslation> = mutableListOf(),
) : AbstractBaseEntity(), ITranslatable<String, CategoryTranslation> {
    override fun toString(): String = "${this::class.simpleName}(id = $id)"
}
